dipy/reconst/utils.py

```diff
@@ -50,6 +50,102 @@ def dki_design_matrix(gtab):
     return B
 
 
+def cti_design_matrix(gtab1, gtab2):
+    r"""Construct B design matrix for CTI.
+
+    Parameters
+    ----------
+    gtab1: dipy.core.gradients.GradientTable
+        A GradientTable class instance for first DDE diffusion epoch
+    gtab2: dipy.core.gradients.GradientTable
+        A GradientTable class instance for second DDE diffusion epoch
+
+    Returns
+    -------
+    B: array(N, 43)
+    Design matrix or B matrix for the CTI model assuming multiple Gaussian Components
+
+    """
+    b1 = gtab1.bvals
+    b2 = gtab2.bvals
+    n1 = gtab1.bvecs
+    n2 = gtab2.bvecs
+    B = np.zeros((len(b1), 43))
+
+    B[:, 0] = -b1 * (n1[:, 0]**2) - b2 * (n2[:, 0]**2)
+    B[:, 1] = -2 * b1 * n1[:, 0] * n1[:, 1] - 2 * b2 * n2[:, 0] * n2[:, 1]
+    B[:, 2] = -b1 * n1[:, 1]**2 - b2 * n2[:, 1]**2
+    B[:, 3] = -2 * b1 * n1[:, 0] * n1[:, 2] + 2 * b2 * n2[:, 0] * n2[:, 2]
+    B[:, 4] = - 2 * b1 * n1[:, 1] * n1[:, 2] + 2 * b2 * n2[:, 1] * n2[:, 2]
```

Comment:
You have here a wrong sign. This code line should be:
B[:, 4] = -2 * b1 * n1[:, 1] * n1[:, 2] - 2 * b2 * n2[:, 1] * n2[:, 2]